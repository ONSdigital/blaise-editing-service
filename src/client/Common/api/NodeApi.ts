import axios from 'axios';
import notFound from '../../../server/helpers/axiosHelper';
import { Survey } from '../../../common/interfaces/surveyInterface';

async function getDataFromNode<T>(url: string, notFoundError: string): Promise<T> {
  try {
    const response = await axios.get(url);

    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error(notFoundError);
    }
    throw new Error('Unable to complete request, please try again in a few minutes');
  }
}

export default async function getSurveys(): Promise<Survey[]> {
  return getDataFromNode('/api/surveys', 'Unable to find surveys, please contact Richmond Rice');
}
