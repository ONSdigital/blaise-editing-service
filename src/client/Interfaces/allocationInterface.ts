export interface AllocationDetails {
  editors: UserAllocationDetails[]
  interviewers: UserAllocationDetails[]
}

interface UserAllocationDetails {
  name: string;
  Cases: string[]
}
