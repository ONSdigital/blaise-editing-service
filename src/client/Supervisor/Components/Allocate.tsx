import { ONSButton, ONSPanel, ONSSelect } from 'blaise-design-system-react-components';
import { ReactElement } from 'react';

export default function AllocateCases(): ReactElement {
  return (
    <>
      <ONSPanel status="info">
        Allocate cases from an interviewer to an editor. All cases conducted by that interviewer will be allocated to the editor
      </ONSPanel>
      <br />
      <ONSSelect
        defaultValue="Doctor Doom"
        id="select-editor"
        label="Select editor"
        options={[
          {
            label: 'Doctor Doom',
            value: 'ddoom',
          },
          {
            label: 'Captain Fantastic',
            value: 'cfantastic',
          },
          {
            label: 'The Thing',
            value: 'tthing',
          },
        ]}
        value=""
      />

      <ONSSelect
        defaultValue="Interviewer1"
        id="select-interviewer"
        label="Select interviewer"
        options={[
          {
            label: 'Interviewer 1 (10 cases)',
            value: 'interviewer1',
          },
          {
            label: 'Interviewer 2 (20 cases)',
            value: 'interviewer2',
          },
          {
            label: 'Interviewer 3 (20 cases)',
            value: 'interviewer3',
          },
          {
            label: 'Interviewer 4 (5 cases)',
            value: 'interviewer4',
          },
          {
            label: 'Interviewer 5 (20 cases)',
            value: 'interviewer5',
          },
        ]}
        value=""
      />
      <br />
      <ONSButton
        label="Allocate"
        primary
      />

    </>
  );
}
