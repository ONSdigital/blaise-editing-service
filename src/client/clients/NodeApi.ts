import axios from 'axios';
import { CaseDetails, CaseFactsheetDetails } from '../../common/interfaces/caseInterface';
import notFound from '../../common/helpers/axiosHelper';
import { Survey } from '../../common/interfaces/surveyInterface';

/* eslint-disable class-methods-use-this */
export default class NodeApi {
  constructor() {
    this.getSurveys = this.getSurveys.bind(this);
    this.getCases = this.getCases.bind(this);
    this.getCaseFactsheet = this.getCaseFactsheet.bind(this);
  }

  async getDataFromNode<T>(url: string, notFoundError: string): Promise<T> {
    try {
      const response = await axios.get(url);

      return response.data;
    } catch (error) {
      if (notFound(error)) {
        throw new Error(notFoundError);
      }
      throw new Error('Unable to complete request, please try again in a few minutes');
    }
  }

  async getSurveys(): Promise<Survey[]> {
    return this.getDataFromNode('/api/surveys', 'Unable to find surveys, please contact Richmond Rice');
  }

  async getCases(questionnaireName: string): Promise<CaseDetails[]> {
    return this.getDataFromNode(`/api/questionnaires/${questionnaireName}/cases`, 'The questionnaire is no longer available');
  }

  async getCaseFactsheet(questionnaireName: string, caseId: string): Promise<CaseFactsheetDetails> {
    return this.getDataFromNode(`/api/questionnaires/${questionnaireName}/cases/${caseId}/factsheet`, 'The questionnaire is no longer available');
  }
}
