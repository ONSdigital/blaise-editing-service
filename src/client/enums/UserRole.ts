enum UserRole {
  Manager = 'Manager',
  Editor = 'Editor',
}

export default UserRole;
