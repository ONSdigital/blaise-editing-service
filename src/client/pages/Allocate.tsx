import { useParams } from 'react-router-dom';
import { AllocationDetails } from '../../common/interfaces/surveyInterface';
import { getAllocationDetails } from '../api/NodeApi';
import { useAsyncRequestWithParam } from '../hooks/useAsyncRequest';
import AsyncContent from '../components/AsyncContent';
import QuestionnaireAllocation from '../components/QuestionnaireAllocation';

function DisplayAllocation(questionnaireName:string) {
  const allocationDetails = useAsyncRequestWithParam<AllocationDetails, string>(getAllocationDetails, questionnaireName);

  return (
    <div data-testid="Allocate">
      <AsyncContent content={allocationDetails}>
        {(loadedAllocationDetails) => <QuestionnaireAllocation questionnaireName={questionnaireName} allocationDetails={loadedAllocationDetails} />}
      </AsyncContent>
    </div>
  );
}

export type AllocateParams = {
  questionnaireName: string
};

export default function Allocate() {
  const { questionnaireName } = useParams<keyof AllocateParams>() as AllocateParams;

  return DisplayAllocation(questionnaireName);
}
