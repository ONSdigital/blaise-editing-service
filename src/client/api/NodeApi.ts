import axios from 'axios';

import { CaseEditInformation } from 'blaise-api-node-client/lib/cjs/interfaces/case';
import { User } from 'blaise-api-node-client/lib/cjs/interfaces/user';
import notFound from '../../server/helpers/axiosHelper';
import { Survey } from '../../common/interfaces/surveyInterface';
import { SupervisorInformation } from '../Interfaces/supervisorInterface';
import { EditorInformation } from '../Interfaces/editorInterface';
import mapEditorInformation from '../Mappers/editorInformaitionMapper';
import mapSupervisorInformation from '../Mappers/supervisorInformationMapper';
import { CaseSummaryDetails } from '../../common/interfaces/caseInterface';
import { CasesNotAllocatedInformation } from '../Interfaces/caseAllocationInterface';
import mapCasesNotAllocated from '../Mappers/caseAllocationMapper';
// import { caseSummaryDetailsMockObject } from '../../test/server/mockObjects/CaseMockObject';

async function getDataFromNode<T>(url: string, notFoundError: string): Promise<T> {
  try {
    const response = await axios.get(url);

    return response.data;
  } catch (error) {
    if (notFound(error)) {
      throw new Error(notFoundError);
    }
    throw new Error('Unable to complete request, please try again in a few minutes');
  }
}

export async function getSurveys(userRole: string): Promise<Survey[]> {
  return getDataFromNode(`/api/surveys?userRole=${userRole}`, 'Unable to find surveys, please contact Richmond Rice');
}

export async function getCaseSummary(questionnaireName: string, caseId: string): Promise<CaseSummaryDetails> {
  return getDataFromNode(`/api/questionnaires/${questionnaireName.toUpperCase()}/cases/${caseId}/summary`, 'The questionnaire is no longer available');
}

async function getCaseEditInformation(questionnaireName: string, userRole: string) {
  // TODO Fix the URL upper
  return getDataFromNode<CaseEditInformation[]>(`/api/questionnaires/${questionnaireName.toUpperCase()}/cases/edit?userRole=${userRole}`, 'Unable to find case edit information, please contact Richmond Rice');
}

export async function getEditorInformation(questionnaireName: string, editorUsername:string, editorRole: string): Promise<EditorInformation> {
  const caseEditInformationList = await getCaseEditInformation(questionnaireName, editorRole);
  const caseEditInformationListForEditor = caseEditInformationList.filter((caseEditInformation) => caseEditInformation.assignedTo === editorUsername);

  return mapEditorInformation(caseEditInformationListForEditor);
}

export async function getSupervisorEditorInformation(questionnaireName: string, supervisorRole: string, editorRole: string): Promise<SupervisorInformation> {
  const caseEditInformationList = await getCaseEditInformation(questionnaireName, supervisorRole);
  const editors = await getDataFromNode<User[]>(`/api/users?userRole=${editorRole}`, 'Unable to find user information, please contact Richmond Rice');

  return mapSupervisorInformation(caseEditInformationList, editors);
}

export async function getCasesNotAllocatedInformation(questionnaireName: string, supervisorRole: string, editorRole: string): Promise<CasesNotAllocatedInformation> {
  const caseEditInformationList = await getCaseEditInformation(questionnaireName, supervisorRole);
  const editors = await getDataFromNode<User[]>(`/api/users?userRole=${editorRole}`, 'Unable to find user information, please contact Richmond Rice');

  return mapCasesNotAllocated(caseEditInformationList, editors);
}
