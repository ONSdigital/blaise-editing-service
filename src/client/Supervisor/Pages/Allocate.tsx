import { ReactElement, useState } from 'react';
import { useParams } from 'react-router-dom';
import { ONSPanel } from 'blaise-design-system-react-components';
import UserRole from '../../Common/enums/UserRole';
import ErrorPanel from '../../Common/components/ErrorPanel';
import SuccessPanel from '../../Common/components/SuccessPanel';
import AllocateContent from '../Components/AllocateContent';
import { Message } from '../../Common/types/MessageType';

interface AllocateProps {
  supervisorRole: UserRole;
  editorRole: UserRole;
  reallocate: boolean;
}

export type AllocateParams = {
  questionnaireName: string
};

export default function Allocate({ supervisorRole, editorRole, reallocate } : AllocateProps): ReactElement {
  const { questionnaireName } = useParams<keyof AllocateParams>() as AllocateParams;
  const defaultMessage: Message = { show: false, text: '', type: '' };
  const [message, setMessage] = useState(defaultMessage);

  return (
    <>
      <ONSPanel status="info" testID="allocation-page-panel">
        {
          reallocate === false
            ? 'Allocate cases from an interviewer to an editor. All cases conducted by that interviewer will be allocated to the editor'
            : 'Reallocate cases from one editor to another editor. All non-completed cases will be transfered'
        }
      </ONSPanel>

      {message.show && message.type === 'error' && <ErrorPanel message={message.text} setMessage={setMessage} /> }
      {message.show && message.type === 'success' && <SuccessPanel message={message.text} setMessage={setMessage} /> }

      <br />
      <h3>{questionnaireName}</h3>

      <AllocateContent questionnaireName={questionnaireName} supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} setMessage={setMessage} />
    </>
  );
}
