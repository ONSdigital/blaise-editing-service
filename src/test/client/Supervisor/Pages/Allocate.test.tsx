import { render, act, RenderResult } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { getAllocationDetails } from '../../../../client/api/NodeApi';
import UserRole from '../../../../client/Common/enums/UserRole';
import { AllocationDetails } from '../../../../common/interfaces/allocationInterface';
import AllocationMockObject from '../../MockObjects/AllocationMockObjects';
import Allocate from '../../../../client/Supervisor/Pages/Allocate';

// set global vars
const supervisorRole:UserRole = UserRole.SVT_Supervisor;
const editorRole:UserRole = UserRole.SVT_Editor;
let view:RenderResult;

// set mocks
jest.mock('../../../../client/api/NodeApi');
const getAllocationDetailsMock = getAllocationDetails as jest.Mock<Promise<AllocationDetails>>;

describe('Given we wish to allocte cases from an Interviewer to an Editor', () => {
  const reallocate = false;

  beforeEach(() => {
    getAllocationDetailsMock.mockReturnValueOnce(Promise.resolve(AllocationMockObject));
  });

  afterEach(() => {
    getAllocationDetailsMock.mockReset();
  });

  it('should render the allocation page correctly', async () => {
    // arrange

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    expect(view).toMatchSnapshot();
  });

  it('should display an have the correct page info for allocation', async () => {
    // arrange

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const panelView = view.getByTestId('allocation-page-panel');
    expect(panelView).toHaveTextContent('Allocate cases from an interviewer to an editor. All cases conducted by that interviewer will be allocated to the editor');
  });

  it('should display alist of availiable interviewer for alloction from', async () => {
    // arrange

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const interviewerListOption = view.getByTestId('select-from');
    expect(interviewerListOption.childElementCount).toEqual(AllocationMockObject.Interviewers.length+1);

    expect(interviewerListOption.children[0]).toHaveTextContent("Select an option");
    AllocationMockObject.Interviewers.forEach((interviewer, interviewerIndex) => {
      expect(interviewerListOption.children[interviewerIndex+1]).toHaveTextContent(`${interviewer.Name} (${interviewer.Cases.length} case(s))`);
    });
  });

  it('should display alist of availiable editors for alloction to', async () => {
    // arrange

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const editorListOption = view.getByTestId('select-to');
    expect(editorListOption.childElementCount).toEqual(AllocationMockObject.Editors.length+1);

    expect(editorListOption.children[0]).toHaveTextContent("Select an option");
    AllocationMockObject.Editors.forEach((Editor, editorIndex) => {
      expect(editorListOption.children[editorIndex+1]).toHaveTextContent(`${Editor.Name} (${Editor.Cases.length} case(s))`);
    });
  });
});

describe('Given we wish to reallocte cases from an Editor to another Editor', () => {
  const reallocate = true;

  beforeEach(() => {
    getAllocationDetailsMock.mockReturnValueOnce(Promise.resolve(AllocationMockObject));
  });

  afterEach(() => {
    getAllocationDetailsMock.mockReset();
  });

  it('should render the allocation page correctly', async () => {
    // arrange

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    expect(view).toMatchSnapshot();
  });

  it('should display an have the correct page info for reallocation', async () => {
    // arrange

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const panelView = view.getByTestId('allocation-page-panel');
    expect(panelView).toHaveTextContent('Reallocate cases from one editor to another editor. All non-completed cases will be transfered');
  });

  it('should display alist of editors for realloction from', async () => {
    // arrange

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const editorListOption = view.getByTestId('select-from');
    expect(editorListOption.childElementCount).toEqual(AllocationMockObject.Editors.length+1);

    expect(editorListOption.children[0]).toHaveTextContent("Select an option");
    AllocationMockObject.Editors.forEach((editor, editorIndex) => {
      expect(editorListOption.children[editorIndex+1]).toHaveTextContent(`${editor.Name} (${editor.Cases.length} case(s))`);
    });
  });

  it('should display a list of availiable editors for realloction to', async () => {
    // arrange

    // act
    await act(async () => {
      view = render(
        <BrowserRouter>
          <Allocate supervisorRole={supervisorRole} editorRole={editorRole} reallocate={reallocate} />
        </BrowserRouter>,
      );
    });

    // assert
    const editorListOption = view.getByTestId('select-to');
    expect(editorListOption.childElementCount).toEqual(AllocationMockObject.Editors.length+1);

    expect(editorListOption.children[0]).toHaveTextContent("Select an option");
    AllocationMockObject.Editors.forEach((Editor, editorIndex) => {
      expect(editorListOption.children[editorIndex+1]).toHaveTextContent(`${Editor.Name} (${Editor.Cases.length} case(s))`);
    });
  });
});
