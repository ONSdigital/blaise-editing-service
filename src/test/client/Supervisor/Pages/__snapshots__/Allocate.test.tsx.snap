// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given we wish to allocte cases from an Interviewer to an Editor should render the allocation page correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
        data-testid="allocation-page-panel"
      >
        <div
          class="ons-panel__body "
        >
          Allocate cases from an interviewer to an editor. All cases conducted by that interviewer will be allocated to the editor
        </div>
      </div>
      <br />
      <h3 />
      <div>
        <label
          class="ons-label"
          for="select-from"
        >
          Allocate cases from
           
        </label>
        <select
          class="ons-input "
          data-testid="select-from"
          id="select-from"
          name="select"
        >
          <option
            data-testid="select-select-from"
            disabled=""
            selected=""
            value=""
          >
            Select an option
          </option>
          <option
            data-testid="option-select-from-bob"
            value="bob"
          >
            bob (2 case(s))
          </option>
          <option
            data-testid="option-select-from-jamester"
            value="jamester"
          >
            jamester (1 case(s))
          </option>
        </select>
      </div>
      <div>
        <label
          class="ons-label"
          for="select-to"
        >
          To
           
        </label>
        <select
          class="ons-input "
          data-testid="select-to"
          id="select-to"
          name="select"
        >
          <option
            data-testid="select-select-to"
            disabled=""
            selected=""
            value=""
          >
            Select an option
          </option>
          <option
            data-testid="option-select-to-Dave"
            value="Dave"
          >
            Dave (0 case(s))
          </option>
          <option
            data-testid="option-select-to-Jake"
            value="Jake"
          >
            Jake (2 case(s))
          </option>
          <option
            data-testid="option-select-to-Rich"
            value="Rich"
          >
            Rich (0 case(s))
          </option>
        </select>
      </div>
      <br />
      <button
        class="ons-btn "
        data-testid="button"
        type="button"
      >
        <span
          class="ons-btn__inner"
        >
          Allocate
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
      data-testid="allocation-page-panel"
    >
      <div
        class="ons-panel__body "
      >
        Allocate cases from an interviewer to an editor. All cases conducted by that interviewer will be allocated to the editor
      </div>
    </div>
    <br />
    <h3 />
    <div>
      <label
        class="ons-label"
        for="select-from"
      >
        Allocate cases from
         
      </label>
      <select
        class="ons-input "
        data-testid="select-from"
        id="select-from"
        name="select"
      >
        <option
          data-testid="select-select-from"
          disabled=""
          selected=""
          value=""
        >
          Select an option
        </option>
        <option
          data-testid="option-select-from-bob"
          value="bob"
        >
          bob (2 case(s))
        </option>
        <option
          data-testid="option-select-from-jamester"
          value="jamester"
        >
          jamester (1 case(s))
        </option>
      </select>
    </div>
    <div>
      <label
        class="ons-label"
        for="select-to"
      >
        To
         
      </label>
      <select
        class="ons-input "
        data-testid="select-to"
        id="select-to"
        name="select"
      >
        <option
          data-testid="select-select-to"
          disabled=""
          selected=""
          value=""
        >
          Select an option
        </option>
        <option
          data-testid="option-select-to-Dave"
          value="Dave"
        >
          Dave (0 case(s))
        </option>
        <option
          data-testid="option-select-to-Jake"
          value="Jake"
        >
          Jake (2 case(s))
        </option>
        <option
          data-testid="option-select-to-Rich"
          value="Rich"
        >
          Rich (0 case(s))
        </option>
      </select>
    </div>
    <br />
    <button
      class="ons-btn "
      data-testid="button"
      type="button"
    >
      <span
        class="ons-btn__inner"
      >
        Allocate
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Given we wish to reallocte cases from an Editor to another Editor should render the allocation page correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
        data-testid="allocation-page-panel"
      >
        <div
          class="ons-panel__body "
        >
          Reallocate cases from one editor to another editor. All non-completed cases will be transfered
        </div>
      </div>
      <br />
      <h3 />
      <div>
        <label
          class="ons-label"
          for="select-from"
        >
          Allocate cases from
           
        </label>
        <select
          class="ons-input "
          data-testid="select-from"
          id="select-from"
          name="select"
        >
          <option
            data-testid="select-select-from"
            disabled=""
            selected=""
            value=""
          >
            Select an option
          </option>
          <option
            data-testid="option-select-from-Dave"
            value="Dave"
          >
            Dave (0 case(s))
          </option>
          <option
            data-testid="option-select-from-Jake"
            value="Jake"
          >
            Jake (2 case(s))
          </option>
          <option
            data-testid="option-select-from-Rich"
            value="Rich"
          >
            Rich (0 case(s))
          </option>
        </select>
      </div>
      <div>
        <label
          class="ons-label"
          for="select-to"
        >
          To
           
        </label>
        <select
          class="ons-input "
          data-testid="select-to"
          id="select-to"
          name="select"
        >
          <option
            data-testid="select-select-to"
            disabled=""
            selected=""
            value=""
          >
            Select an option
          </option>
          <option
            data-testid="option-select-to-Dave"
            value="Dave"
          >
            Dave (0 case(s))
          </option>
          <option
            data-testid="option-select-to-Jake"
            value="Jake"
          >
            Jake (2 case(s))
          </option>
          <option
            data-testid="option-select-to-Rich"
            value="Rich"
          >
            Rich (0 case(s))
          </option>
        </select>
      </div>
      <br />
      <button
        class="ons-btn "
        data-testid="button"
        type="button"
      >
        <span
          class="ons-btn__inner"
        >
          Allocate
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
      data-testid="allocation-page-panel"
    >
      <div
        class="ons-panel__body "
      >
        Reallocate cases from one editor to another editor. All non-completed cases will be transfered
      </div>
    </div>
    <br />
    <h3 />
    <div>
      <label
        class="ons-label"
        for="select-from"
      >
        Allocate cases from
         
      </label>
      <select
        class="ons-input "
        data-testid="select-from"
        id="select-from"
        name="select"
      >
        <option
          data-testid="select-select-from"
          disabled=""
          selected=""
          value=""
        >
          Select an option
        </option>
        <option
          data-testid="option-select-from-Dave"
          value="Dave"
        >
          Dave (0 case(s))
        </option>
        <option
          data-testid="option-select-from-Jake"
          value="Jake"
        >
          Jake (2 case(s))
        </option>
        <option
          data-testid="option-select-from-Rich"
          value="Rich"
        >
          Rich (0 case(s))
        </option>
      </select>
    </div>
    <div>
      <label
        class="ons-label"
        for="select-to"
      >
        To
         
      </label>
      <select
        class="ons-input "
        data-testid="select-to"
        id="select-to"
        name="select"
      >
        <option
          data-testid="select-select-to"
          disabled=""
          selected=""
          value=""
        >
          Select an option
        </option>
        <option
          data-testid="option-select-to-Dave"
          value="Dave"
        >
          Dave (0 case(s))
        </option>
        <option
          data-testid="option-select-to-Jake"
          value="Jake"
        >
          Jake (2 case(s))
        </option>
        <option
          data-testid="option-select-to-Rich"
          value="Rich"
        >
          Rich (0 case(s))
        </option>
      </select>
    </div>
    <br />
    <button
      class="ons-btn "
      data-testid="button"
      type="button"
    >
      <span
        class="ons-btn__inner"
      >
        Allocate
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
